ethcore/src/state/mod.rs

```diff
@@ -186,8 +195,8 @@ impl AccountEntry {
 /// checkpoint can be discateded with `discard_checkpoint`. All of the orignal
 /// backed-up values are moved into a parent checkpoint (if any).
 ///
-pub struct State {
-	db: StateDB,
+pub struct State<B> {
```

Comment:
That's the same commit as #4632 as this builds upon that, we can ice this until it's in.